arangod/Transaction/Methods.cpp

```diff
@@ -2033,6 +2043,16 @@ OperationResult transaction::Methods::modifyLocal(std::string const& collectionN
       if (!options.isSynchronousReplicationFrom.empty()) {
         return OperationResult(TRI_ERROR_CLUSTER_SHARD_LEADER_REFUSES_REPLICATION);
       }
+      if (followerInfo->get()->size() + 1 < collection->minReplicationFactor()) {
+        // We cannot fulfill minimum replication Factor.
+        // Reject write.
+        LOG_TOPIC("2e35a", ERR, Logger::REPLICATION)
+            << "Less then minReplicationFactor "
+            << basics::StringUtils::itoa(collection->minReplicationFactor())
+            << " followers in sync. Shard  " << collection->name()
+            << " is temporarily in read-only mode.";
+        return OperationResult(TRI_ERROR_ARANGO_READ_ONLY, options);
+      }
```

Comment:
Is there a central place for checking this? Is there any operation that can be carried out, even with fewer followers? I don't think so,

Reply:
This is as central as it gets currently.

Reply:
Also i do not think so right now.